services/wallet/transfer/helpers.go

```diff
@@ -0,0 +1,245 @@
+			&fromTxHash,
+			&multiTransaction.FromAddress,
+			&multiTransaction.FromAsset,
+			&fromAmountDB,
+			&multiTransaction.ToNetworkID,
+			&toTxHash,
+			&multiTransaction.ToAddress,
+			&multiTransaction.ToAsset,
+			&toAmountDB,
+			&multiTransaction.Type,
+			&multiTransaction.CrossTxID,
+			&multiTransaction.Timestamp,
+		)
+		if len(fromTxHash) > 0 {
+			multiTransaction.FromTxHash = common.BytesToHash(fromTxHash)
+		}
+		if len(toTxHash) > 0 {
+			multiTransaction.ToTxHash = common.BytesToHash(toTxHash)
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if fromAmountDB.Valid {
+			multiTransaction.FromAmount = new(hexutil.Big)
+			if _, ok := (*big.Int)(multiTransaction.FromAmount).SetString(fromAmountDB.String, 0); !ok {
+				return nil, errors.New("failed to convert fromAmountDB.String to big.Int: " + fromAmountDB.String)
+			}
+		}
+
+		if toAmountDB.Valid {
+			multiTransaction.ToAmount = new(hexutil.Big)
+			if _, ok := (*big.Int)(multiTransaction.ToAmount).SetString(toAmountDB.String, 0); !ok {
+				return nil, errors.New("failed to convert fromAmountDB.String to big.Int: " + toAmountDB.String)
+			}
+		}
+
+		multiTransactions = append(multiTransactions, multiTransaction)
+	}
+
+	return multiTransactions, nil
+}
+
+func (tm *TransactionManager) ensureTransactionsSigned(signatures map[string]SignatureDetails) error {
+	if len(tm.transactionsForKeycardSingning) == 0 {
+		return errors.New("no transactions to proceed with")
+	}
+	if len(signatures) != len(tm.transactionsForKeycardSingning) {
+		return errors.New("not all transactions have been signed")
+	}
+
+	// check if all transactions have been signed
+	for hash, desc := range tm.transactionsForKeycardSingning {
+		sigDetails, ok := signatures[hash.String()]
+		if !ok {
+			return fmt.Errorf("missing signature for transaction %s", hash)
+		}
+
+		rBytes, _ := hex.DecodeString(sigDetails.R)
+		sBytes, _ := hex.DecodeString(sigDetails.S)
+		vByte := byte(0)
+		if sigDetails.V == "01" {
+			vByte = 1
+		}
+
+		desc.signature = make([]byte, crypto.SignatureLength)
+		copy(desc.signature[32-len(rBytes):32], rBytes)
+		copy(desc.signature[64-len(rBytes):64], sBytes)
+		desc.signature[64] = vByte
+	}
+
+	return nil
+}
+
+func multiTransactionFromCommand(command *MultiTransactionCommand) *MultiTransaction {
+	multiTransaction := NewMultiTransaction(
+		/* Timestamp:     */ uint64(time.Now().Unix()),
+		/* FromNetworkID: */ 0,
+		/* ToNetworkID:	  */ 0,
+		/* FromTxHash:    */ common.Hash{},
+		/* ToTxHash:      */ common.Hash{},
+		/* FromAddress:   */ command.FromAddress,
+		/* ToAddress:     */ command.ToAddress,
+		/* FromAsset:     */ command.FromAsset,
+		/* ToAsset:       */ command.ToAsset,
+		/* FromAmount:    */ command.FromAmount,
+		/* ToAmount:      */ new(hexutil.Big),
+		/* Type:		  */ command.Type,
+		/* CrossTxID:	  */ "",
+	)
+
+	return multiTransaction
+}
+
+func (tm *TransactionManager) buildTransactions(bridges map[string]bridge.Bridge) ([]string, error) {
```

Comment:
@IvanBelyakoff I personally wouldn't move functions that refer to certain types like `func (tm *TransactionManager) ...` to the file named "helpers". Please check if there is a chance to pull them out from the `TransactionManager` and keep them here or move them back.